flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1287,13 +1287,17 @@ object ScalarOperatorGens {
            |boolean $nullTerm;
            |if (${condition.resultTerm}) {
            |  ${trueAction.code}
-           |  $resultTerm = ${trueAction.resultTerm};
            |  $nullTerm = ${trueAction.nullTerm};
+           |  if (!$nullTerm) {
+           |    $resultTerm = ${trueAction.resultTerm};
```

Comment:
You need assign `resultTerm` to a default value too.
`val defaultValue = primitiveDefaultValue(resultType)`

Reply:
thanks for the reminder. done.